### DISCUSSION TITLE
Resource not getting created when added to chart

### DISCUSSION THREAD

**Discussion by nicogl** [fluxcd/flux2#1161] [Q&A] [answered]
I have a HelmRelease already deployed and running in my cluster. The Chart.yaml file as well as /templates folder containing all Helm resources/manifests is pointed by the HelmRelease manifest.

Then I edited one of the yaml files in the /templates folder to add a new resource (i.e. ClusterRole) and commited the change to the repository.
A reconciliation process is triggered but when I query the ClusterRoles in the cluster, the one I added doesn't appear.
I tried manually running the `flux reconcile` command for the kustomization and helmrelease, but didn't work.

Why this resource change/addition is not being recognized by Flux? Shouldn't it realize about the difference between repository yaml and cluster state and try to syncronize accordingly?

Thanks

**Comment by hiddeco** [comment, marked as answer]
Did you also bump the chart version in your `Chart.yaml`? This is a requirement for changes to be noticed and picked up.

**Reply by nicogl** [reply]
Thanks @hiddeco 

In fact I didn't do it... because I don't know what "_bumping the chart version_" exactly mean. I guess you're referring to this excerpt in the Flux [documentation](https://toolkit.fluxcd.io/guides/helmreleases/): "_**There is one caveat you should be aware of**: to make the source-controller produce a new chart artifact, the version in the Chart.yaml of the chart must be bumped._"

Note that I'm handing my Helm charts with Git repository, not Helm repository, so I have to comply with above statement.
As a side note, I cannot use Helm repository because I'm using AWS ECR which doesn't support it yet.

I tried removing/commenting the version property in the Chart.yaml, like:

`#version: 1.0.0`

but in this case I keep on getting the message: "HelmChart 'flux-system/flux-system-mf-helmrelease' is not ready"

I also tried removing the value of the property, like leaving it empty: 

`version:`

but didn't work either.

What does "bump the version" mean then?

Thanks again!

**Reply by hiddeco** [reply]
Changing the version, e.g. `v0.0.1` ->`v0.0.2`.

**Reply by nicogl** [reply]
Cool! That made the trick! 
Any suggestion on automating the version increasing when changes are made to the chart templates?
Thank you very much!

**Reply by hiddeco** [reply]
The change you are waiting for is https://github.com/fluxcd/source-controller/pull/308, which makes it possible to automatically rebuild the chart (with build metadata added to the SemVer) by configuring a flag.
